itest/lib/createTestBrim.js

```diff
@@ -36,6 +38,25 @@ export default (name: string) => {
       )
     },
 
+    setValue(locator: Locator, value: string) {
+      return appStep(`set input ${locator.css} to ${value}`, () =>
+        app.client
+          .waitForVisible(locator.css)
+          .then(() => app.client.setValue(locator.css, value))
+      )
+    },
+
+    getText(locator: Locator) {
+      return appStep(`get text from ${locator.css}`, async () => {
+        await app.client.waitForVisible(locator.css)
+        return await app.client.getText(locator.css)
+      })
+    },
+
+    clickAppMenuItem(id: string) {
+      return app.mainProcess.emit("spectron:clickAppMenuItem", id)
+    },
```

Comment:
Contrast this with `setValue()` above. Here, this operation is asynchronous. A test is going to have to do its own work to ensure the click has actually performed the expected action before the test moves onto the next action. We just need to be clear and set expectations about what these do. I greatly prefer that the layer used by tests is responsible for ensuring the element is ready to be acted on. I understand in some cases that's hard.

Reply:
Yeah, the test will have to ensure that the expected thing takes place after clicking the application menu.

However, the next line of the test will usually do the waiting I think. For example:
```js
brim.clickAppMenuItem("preferences")
brim.getText(preferencesModal) // in here it could wait for the preferencesModal to appear
```